src/blockchain/types.ts

```diff
@@ -41,6 +41,9 @@ export type TransactionInfo = {
   hash: string;
   memo: string;
   authorizationKind: string;
+  sequenceNumber: number;
+  zkappAccountUpdateIds: number[];
+  zkappEventElementIds: number[];
```

Comment:
You'll want another similar field for Events (this is for Actions) to include as part of the Events data, no?

Reply:
Why do we need that, assuming events will not be sorted?

Reply:
>assuming events will not be sorted

Not sure what you mean by this. Events are recorded in the same order as actions i.e., sequence of blocks, sequence of transactions within a block, sequence of account updates within a transaction and sequence of events within an account update. The only difference is that the actions are accumulated on chain in action state and so one will always want to know the order of actions to validate against action states. For events it is more of a utility where an app might want to process events in the order they were confirmed/included in the chain. As an AP,I it seems like it should be providing that utility